src/navigator.c

```diff
@@ -138,17 +139,22 @@ void imv_navigator_select_rel(struct imv_navigator *nav, int direction)
   } else if(direction < -1) {
     direction = -1;
   } else if(direction == 0) {
-    return;
+    return 0;
   }
 
   nav->cur_path += direction;
   if(nav->cur_path == nav->num_paths) {
+    /* end of list reached */
+    if(!cycle) {
+      return 0;
+    }
     nav->cur_path = 0;
   } else if(nav->cur_path < 0) {
     nav->cur_path = nav->num_paths - 1;
```

Comment:
Oh, yeah thats quite inconsistent. Didn't think about this. I think we should exit too in this case, doing nothing might be confusing. Will add this, ok?